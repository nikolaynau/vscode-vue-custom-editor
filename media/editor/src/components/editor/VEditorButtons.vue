<script setup lang="ts">
import VButton from '../button/VButton.vue';
import type { ButtonDefinition } from './utils/buttons';

export interface Props {
  items?: Array<ButtonDefinition>;
}

withDefaults(defineProps<Props>(), {
  items: () => []
});

const emit = defineEmits<{
  (e: 'click', button: ButtonDefinition): void;
}>();

function onClick(button: ButtonDefinition) {
  emit('click', button);
}

function getLabel(value: number) {
  return value > 0 ? `+${value}` : `${value}`;
}
</script>

<template>
  <VButton v-for="button in items" :key="button.id" @click="onClick(button)">
    {{ getLabel(button.value) }}
  </VButton>
</template>
