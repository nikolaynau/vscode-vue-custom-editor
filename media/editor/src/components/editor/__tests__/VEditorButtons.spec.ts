import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VEditorButtons from '../VEditorButtons.vue';
import type { ButtonDefinition } from '../utils/buttons';

describe('VEditorButtons', () => {
  it('render default props', () => {
    const wrapper = mount(VEditorButtons);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('items prop', () => {
    const items: ButtonDefinition[] = [
      {
        id: 1,
        label: '-10',
        value: -10,
        side: 'left'
      },
      {
        id: 2,
        label: '-5',
        value: -5,
        side: 'left'
      }
    ];
    const wrapper = mount(VEditorButtons, { props: { items } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('click event', async () => {
    const items: ButtonDefinition[] = [
      {
        id: 1,
        label: '-10',
        value: -10,
        side: 'left'
      },
      {
        id: 2,
        label: '-5',
        value: -5,
        side: 'left'
      }
    ];
    const wrapper = mount(VEditorButtons, { props: { items } });
    const buttons = wrapper.findAllComponents({ name: 'VButton' });
    expect(buttons).toHaveLength(2);

    await buttons[0].trigger('click');

    const clickEvent = wrapper.emitted('click') as unknown[][];
    expect(clickEvent).toHaveLength(1);
    expect(clickEvent[0][0]).toEqual(items[0]);
  });
});
