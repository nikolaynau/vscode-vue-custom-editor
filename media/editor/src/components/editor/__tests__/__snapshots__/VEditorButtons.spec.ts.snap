// Vitest Snapshot v1

exports[`VEditorButtons > items prop 1`] = `
"<a class=\\"v-button\\" tabindex=\\"0\\">-10</a>
<a class=\\"v-button\\" tabindex=\\"0\\">-5</a>"
`;

exports[`VEditorButtons > render default props 1`] = `""`;
